web/src/main/java/org/fao/geonet/monitor/link/LinkCheckerDefault.java

```diff
@@ -0,0 +1,23 @@
+package org.fao.geonet.monitor.link;
+
+import org.jdom.Element;
+
+public class LinkCheckerDefault implements LinkCheckerInterface {
+    private String getCheckUrl(Element onlineResource) {
+        return LinkCheckerUtils.parseOnlineResource(onlineResource, LinkCheckerUtils.URL_XPATH);
+    }
+
+    @Override
+    public boolean check(Element onlineResource) {
```

Comment:
I'm happy with that abstraction level too, in terms of the classes.

I respectfully and strongly disagree about the duplication though!  It's more code to understand, fix bugs in, and have diverge in behaviour, even when it shouldn't.  But I'm happy for you to keep it like this if you feel that way.